RIOT-OS/RIOT
# sys/test_utils/result_output: Initial implementations and API of turo

### Contribution description

The latest in an effort to add an abstraction layer for RIOT tests allowing a more stable testing API. 
This focuses more on how results are printed or output.
Using the results_output API will allow a guaranteed way output data, meaning no more needing to adapt regex everytime a printf is changed.
The exact output can be changed depending on the module selected, this means if you don't like it you can just implement a variant that suits your needs.
The eventual goal will be to coordinate this with `riotctrl` so data can be used immediately when writing python based automated tests.

There are currently 4 variant implemented:
- json: used for CI and computer interaction
- txt: used for developer interaction
- check: Only ensuring the output follows a correct structure
- led: An example of using the `turo_t *ctx` to flash LEDs

More information is in the `sys/test_utils/results_output.h` file.

Note: The `test_utils_result_output_json` is ~~not~~ fully json compliant ~~as it contains trailing commas. This is allows simple flushing of information and no need for states. Many parses accept trailing commas such as `rapidjson` python parser~~.

### Testing procedure

`OUTPUT_FORMAT={txt, check, json, led} make all -C tests/turo`

_note: native has "leds" but they don't fit the LED0_PIN define so an assertion error will be thrown because native uses GPIO_UNDEF_

### Issues/PRs references

supersedes #15039
also supersedes #11224
Implementation of a cleaner idea of #10624
